include/beman/net/detail/kqueue_context.hpp

```diff
@@ -0,0 +1,405 @@
+// include/beman/net/detail/kqueue_context.hpp                        -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_NET_DETAIL_QUEUE_CONTEXT
+#define INCLUDED_BEMAN_NET_DETAIL_QUEUE_CONTEXT
+
+// ----------------------------------------------------------------------------
+
+#include "beman/net/detail/event_type.hpp"
+#include "beman/net/detail/io_base.hpp"
+#include <array>
+#include <beman/net/detail/netfwd.hpp>
+#include <beman/net/detail/container.hpp>
+#include <beman/net/detail/context_base.hpp>
+#include <beman/net/detail/sorted_list.hpp>
+#include <cstddef>
+#include <cstdint>
+#include <span>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <poll.h>
+#include <fcntl.h>
+#include <iostream>
+#include <sys/event.h>
+#include <map>
+#include <vector>
+#include <cassert>
+#include <optional>
+
+// ----------------------------------------------------------------------------
+
+namespace beman::net::detail {
+struct kqueue_record;
+struct kqueue_context;
+} // namespace beman::net::detail
+
+// ----------------------------------------------------------------------------
+
+struct beman::net::detail::kqueue_record final {
+    kqueue_record(::beman::net::detail::native_handle_type h) : handle(h) {}
+    ::beman::net::detail::native_handle_type handle;
+    bool                                     blocking{true};
+};
+
+// ----------------------------------------------------------------------------
+
+struct beman::net::detail::kqueue_context final : ::beman::net::detail::context_base {
+    static constexpr size_t event_buffer_size = 10;
```

Comment:
There isn't a guarantee that these names are made available in the global namespace when including the respective C++ version of the C headers:

```suggestion
    static constexpr ::std::size_t event_buffer_size = 10;
```